tests/tape/test_jacobian_tape.py

```diff
@@ -454,6 +482,55 @@ def test_single_expectation_value(self, tol):
         expected = np.array([[-np.sin(y) * np.sin(x), np.cos(y) * np.cos(x)]])
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    def test_single_expectation_value_with_num_params_all(self, tol):
```

Comment:
This and the following test are using the `test_single_expectation_value` case to make sure that parameter selection works fine.